### PR TITLE
Restructure active farm info panel and add sandbox redirects

### PR DESCRIPTION
**Pull request by yivlad**

### CODE REVIEW

File: packages/app/src/features/farm-details/logic/useSandboxFarmRedirect.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { watchChainId } from 'wagmi/actions'
+import { useFarmDetailsParams } from './useFarmDetailsParams'
+
+export function useSandboxFarmRedirect(): void {
+  const config = useConfig()
+  const { address: farmAddress, chainId: farmChainId } = useFarmDetailsParams()
+  const { sandboxChainId, originChainId } = useSandboxState()
+  const navigate = useNavigate()
+
+  useEffect(() => {
```
Comment (EmilMalanczak) [inline packages/app/src/features/farm-details/logic/useSandboxFarmRedirect.ts:15]: maybe we could reuse `useChainSensitive` hook here?

Comment (yivlad) [reply packages/app/src/features/farm-details/logic/useSandboxFarmRedirect.ts:15]: `watchChainId` has convenient API with new `chainId` and `prevChainId`, so decided to stick to it

Comment (EmilMalanczak) [reply packages/app/src/features/farm-details/logic/useSandboxFarmRedirect.ts:15]: so it does with `frozenChainId` as equiv to `prevChainId `

Comment (yivlad) [reply packages/app/src/features/farm-details/logic/useSandboxFarmRedirect.ts:15]: Yeah, probably you're right, implementation of `useChainSensitive` can be adjusted to fit the needs here. Maybe we should make consistent, but let's do it later

**Comment by oskarvu** [comment]
I think it will be better to format TVL as compact, so it fits nicer.

**Review by oskarvu — APPROVED** [review]

File: packages/app/src/features/farm-details/logic/useSandboxFarmRedirect.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { paths } from '@/config/paths'
+import { useSandboxState } from '@/domain/sandbox/useSandboxState'
+import { useEffect } from 'react'
+import { generatePath, useNavigate } from 'react-router-dom'
+import { useConfig } from 'wagmi'
+import { watchChainId } from 'wagmi/actions'
+import { useFarmDetailsParams } from './useFarmDetailsParams'
+
+export function useSandboxFarmRedirect(): void {
```
Comment (krzkaczor) [inline packages/app/src/features/farm-details/logic/useSandboxFarmRedirect.ts:9]: i don't like that very much... this should be a generic mechanism so entering to sandbox mode on any page works. (what about market details page?). This should be improved in the future.

File: packages/app/src/features/farm-details/dialogs/unstake/logic/useUnstakeDialog.ts [hunk 3]
```diff
@@ -44,7 +44,10 @@ export interface ExitFarmSwitchInfo {
 }
 
 export function useUnstakeDialog({ farm, initialToken }: UseStakeDialogParams): UseUnstakeDialogResult {
-  const { timestamp } = useTimestamp()
+  const { timestamp, updateTimestamp } = useTimestamp()
```
Comment (oskarvu) [inline packages/app/src/features/farm-details/dialogs/unstake/logic/useUnstakeDialog.ts:47]: Probably worth to extract to separate hook in future. `useNowTimestamp` or something like this.

**Review by EmilMalanczak — APPROVED** [review]

**Review by oskarvu — APPROVED** [review]

**Pull request merged**